lib/assert.js

```diff
@@ -81,8 +81,6 @@ x.throws = function (fn, err, msg) {
 				x.throws(function () {
 					if (fnErr instanceof Error) {
 						throw fnErr;
-					} else {
```

Comment:
Remove the if-guard as well, simply `throw fnError`. This makes it equivalent to `t.throws(() => { throw 'foo' })`.